src/application_settings/_private/file_operations.py

```diff
@@ -89,31 +82,25 @@ def save(path: Path, data: dict[str, Any]) -> None:
         throw_if_file_not_found=False,
         create_file_if_not_found=True,
     ):
-        if (ext := path.suffix[1:].lower()) == FileFormat.JSON.value:
-            return _save_json(path, data)
-        if ext == FileFormat.TOML.value:
-            return _save_toml(path, data)
+        if saver := _get_saver(ext=path.suffix[1:].lower()):
+            return saver(path, data)
     return None
 
 
-def _load_toml(path: Path) -> dict[str, Any]:
-    data_stored: dict[str, Any] = {}
-    with path.open(mode="rb") as fptr:
-        data_stored = tomllib.load(fptr)
-    return data_stored
-
-
-def _load_json(path: Path) -> dict[str, Any]:
-    data_stored: dict[str, Any] = {}
-    with path.open(mode="r") as fptr:
-        data_stored = json.load(fptr)
-    return data_stored
+def _get_loader(ext: str) -> LoaderOpt:
```

Comment:
I think it's a bit cleaner if the get methods would get a `Path` type and split of the extension (the `path.suffix[1:].lower`) themselves.